app/src/main/scala/com/coralogix/operator/logic/operators/coralogixlogger/FailedProvisions.scala

```diff
@@ -23,9 +23,11 @@ class FailedProvisions(failedProvisions: TMap[String, Option[String]]) {
                    entry <- failedProvisions.get(name)
                    result = entry match {
                               case Some(failedResourceVersion) =>
-                                resource.metadata.flatMap(
-                                  _.resourceVersion
-                                ) == failedResourceVersion
+                                resource.metadata
+                                  .flatMap(
+                                    _.resourceVersion
+                                  )
+                                  .toOption == failedResourceVersion
```

Comment:
Is this comparing `Option[T]` to `T`?

Reply:
LHS is `Optional[String]` converted to `Option[String]`, RHS is `Option[String]` because `entry` is the result to get from `TMap` and it's an `Option[Option[String]]`. I can see this is a bit misleading :D

Reply:
Ah hehe alright 👍